django/db/models/fields/__init__.py

```diff
@@ -335,12 +335,15 @@ def _check_null_allowed_for_primary_keys(self):
         else:
             return []
 
-    def _check_backend_specific_checks(self, **kwargs):
+    def _check_backend_specific_checks(self, databases=None, **kwargs):
         app_label = self.model._meta.app_label
-        for db in connections:
-            if router.allow_migrate(db, app_label, model_name=self.model._meta.model_name):
-                return connections[db].validation.check_field(self, **kwargs)
-        return []
+        # In case no database provided(e.g. python manage.py check), check the default
+        databases = databases or ['default']
```

Comment:
These lines should be removed; it's possible for `databases` to be an empty list during tests and when no `--database` is passed to `manage.py` check as you've mentioned. When this happens this check should be entirely skipped.